operator/initc/internal/wait.go

```diff
@@ -94,17 +115,32 @@ func NewPodCliqueState(podCliqueDependencies map[string]int, log logr.Logger) (*
 	return state, nil
 }
 
-// WaitForReady waits for all upstream start-up dependencies to be ready.
+// WaitForReady waits for all parent PodCliques to reach their minimum ready replica count.
+// It sets up Kubernetes informers to watch pod events and blocks until all dependencies are satisfied
+// or the provided context is cancelled.
 func (c *ParentPodCliqueDependencies) WaitForReady(ctx context.Context, log logr.Logger) error {
-	defer close(c.allReadyCh) // Close the channel the informers write to *after* the context they use is cancelled.
-
-	log.Info("Parent PodClique(s) being waited on", "pclqFQNToMinAvailable", c.pclqFQNToMinAvailable)
-
+	// Create Kubernetes client for accessing the API
 	client, err := createClient()
 	if err != nil {
 		return err
 	}
+	return c.WaitForReadyWithClient(ctx, client, log)
+}
+
+// WaitForReadyWithClient waits for all parent PodCliques to reach their minimum ready replica count
+// using the provided Kubernetes client. This enables testing by allowing client injection.
+func (c *ParentPodCliqueDependencies) WaitForReadyWithClient(ctx context.Context, client kubernetes.Interface, log logr.Logger) error {
```

Comment:
instead of creating this function, create the client when constructing `ParentPodCliqueDependencies` and make it part of the struct itself.